### ISSUE TITLE
`no-nonstandard-*-properties` rules does not detect methods on redefinable variables

### ISSUE THREAD

**Issue by zloirock** [eslint-community/eslint-plugin-es-x#218]
```js
let array1 = [1, 2, 3];
array1.addAll(3, 4); // that trigger an `no-nonstandard-*-properties` error
doSomethingWith(array1);
// but
let array2 = [1, 2, 3];
array2.deleteAll(3, 4); // it's not detected by `no-nonstandard-*-properties`
doSomethingWith(array2);
array2 = [1, 2, 3, 4]; // redefined
array2.deleteAll(3, 5); // it's not detected by `no-nonstandard-*-properties` too
doSomethingWith(array2);
```
